extensions.html

```diff
@@ -334,6 +334,117 @@ <h2><dfn>GamepadPose</dfn> Interface</h2>
+      
+      <pre class="idl">
+        enum GamepadLightIndicatorType {
+          "on-off",
+          "rgb"
+        };
+      </pre>
+      
+      <dl data-dfn-for="GamepadLightIndicatorType">
+        <dt>"<dfn>on-off</dfn>"</dt>
+        <dd>
+          Light indicator that supports a single color.
+        </dd>
+        
+        <dt>"<dfn>rgb</dfn>"</dt>
+        <dd>
+          Light indicator that supports multiple colors.
+        </dd>
+      </dl>
+    </section>
+    
+    <section>
+      <h2><dfn>GamepadLightColor</dfn></h2>
+      <p>
+        This represents the color of a light indicator. The default value should
+        be set to 0 for red, green, and blue.
+      </p>
+    
+      <pre class="idl">
+        dictionary GamepadLightColor {
+          required octet red;
+          required octet green;
+          required octet blue;
+       };
+      </pre>
+      
+      <dl data-dfn-for="GamepadLightColor">
+        <dt><dfn>red</dt></dfn>
+        <dd>
+          Red component of the light color, or non-zero value for an
+          <code>on-off</code> light indicator that indicates ON.
+        </dd>
+        
+        <dt><dfn>green</dt></dfn>
+        <dd>
+          Blue component of the light color, or non-zero value for an
+          <code>on-off</code> light indicator that indicates ON.
+        </dd>
+        
+        <dt><dfn>blue</dt></dfn>
+        <dd>
+          Green component of the light color, or non-zero value for an
+          <code>on-off</code> light indicator that indicates ON.
+        </dd>
+      </dl>
+    </section>
+    
+    <section>
+      <h2><dfn>GamepadLightIndicator</dfn></h2>
+      <p>
+        This interface defines a light indicator.
+      </p>
+      
+      <pre class="idl">
+        [Exposed=Window, SecureContext]
+        interface GamepadLightIndicator {
+          readonly attribute GamepadLightIndicatorType type;
+        
+          Promise&lt;long&gt; setColor(GamepadLightColor color);
+      };
+      </pre>
+      
+      <dl data-dfn-for="GamepadLightIndicator">
+        <dt><dfn>type</dt></dfn>
+        <dd>
+          Type of light indicator supported by the gamepad.
+        </dd>
+        
+        <dt><dfn>setColor</dt></dfn>
+        <dd>
+          <p>
+          Sets the color of a light indicator, or sets the light indicator to ON
+          or OFF.
+          </p>
+          
+          <p>
+          For <code>on-off</code> light indicators, one or more non zero values
+          denotes the light indicator is ON, whereas a zero value for all
+          components indicates the light indicator is OFF.
+          </p>
+          
+          <p>
+          For RGB light indicators, one or more non zero values specify the
+          color of the light indicator, whereas all zero values indicate the
+          light indicator is OFF.
+          </p>
+          
+          <p>
+          The returned Promise will resolve <code>true</code> when setting the
```

Comment:
I don't know if this can be guaranteed to succeed. @nondebug, do you have an opinion about this?

Reply:
We typically don't get a signal that the color was changed successfully so we shouldn't claim that in the spec. I think we should reword this so we resolve when the LED color change command is sent and reject if the command couldn't be sent.

Reply:
Ok, so yeah... it shouldn't resolve with `true` either. 

We probably need to rewrite this whole section. I'd suggest having a look at:
https://w3c.github.io/web-share/#share-method

Basically, we need to do a few checks that can cause the promise to reject... the controller has been disconnected/no longer available, etc. There might be a SecurityError if the permission has been changed before `.setColor()` is called. Also, what should happen if:

```
while (true) {
   x.setColor(randomRGG());
}
```

Rate limit? Drop requests on the floor?

Reply:
We discussed this and couldn't come up with any compelling reasons to keep the Promise. It's unlikely that we'll receive any error indication from the device and there isn't any way we can lose permission for a Gamepad. Let's remove the promise and treat setColor as fire-and-forget.